spec/index.html

```diff
@@ -10424,7 +10424,205 @@ <h3>Evaluation Semantics</h3>
+              <p>This process is applied throughout the graph pattern of <code>EXISTS</code>:</p>
+              <div class="defn">
+                <b>Definition: <span id="defn_varrename" name="defn_varrename">Variable Remapping</span></b>
+                <p>
+                  For any algebra expression `X`, define the Variable Remapping `PrjMap(X)` 
+                  of algebra expression `X`:
+                </p>
+                  <pre>PrjMap(X) = replace all project operations Project(P, PV) 
+                with ProjectMap(P, PV) for each projection in X.</pre>
+              </div>
+              <p>
+                The outcome of `PrjMap` is independent of the order of replacement
+                (e.g. bottom-up or top-down).
+                Replacements may happen several times, depending on recursive order
+                but each time a replacement is made, the variable not used anywhere else.
+              </p>
+
+              <div class="note">
+                <p>
+                  A variable inside a project expression that is not in the variables projected
+                  is not affected by the values insertion operation because it is renamed apart.
+                </p>
+                <p>
+                  This operation is as part of the <a href="#sparqlQuery">translation to the SPARQL
+                  algebra</a>.
+                </p>
+              </div>
+            </section>
+            <section>
+              <h4>Values Insertion</h4>
+              <p>
+                Alternative 1: rewrite the algebra during 
+                <a href="#sparqlQuery">translation to the SPARQL algebra</a>
+                to include a function that evaluates to the current row.
+              </p>
+
+              <div class="defn">
+                <div>
+                  <b>Definition: <span id="defn_correlate" name="defn_correlate">Values In Scope</span></b>
+                  <p>
+                    Define the function `BindingInScope()`. 
+                    Evaluation of `BindingInScope()` results in a table of one row,
+                    being the current binding of the enclosing filter.
+                  </p>
+                </div>
+              </div>
+
+              <div class="defn">
+                <div>
+                  <b>Definition: Access to the current binding</b>
+                  <p>
+                    During <a href="#sparqlQuery">translation to the SPARQL algebra</a>
+                  </p>
+                  <pre>
+Replace each occurence of `Y` in X where `Y` is one of
+          <a href="#sparqlTranslateBasicGraphPatterns">Basic Graph Pattern</a>,
+          <a href="#sparqlTranslatePathExpressions">Property Path Expression</a>,
+          <a href="#sparqlTranslateGraphPatterns">`Graph(Var, pattern)`</a>,
+          <a href="#https://www.w3.org/TR/sparql12-query/#sparqlTranslateGraphPatterns">Inline Data</a>
+          with `join(Y, BindingInScope())`.</pre>
```

Comment:
1. What is X?
2. Instead of Property Path Expression, do you actually mean Property Path Pattern?